pims/base_frames.py

```diff
@@ -528,3 +528,268 @@ def process(img_or_iterable, *args, **kwargs):
                        "other objects, its behavior is "
                        "unchanged.\n\n") + process.__doc__
     return process
+
+
+class FramesDimension(object):
```

Comment:
Might be worth looking at http://boltons.readthedocs.org/en/latest/namedutils.html#boltons.namedutils.namedlist which is mutable `namedtuple`. It will be lighter weight at runtime and less code for use to maintain.